test/runnable/testcontracts.d

```diff
@@ -402,6 +402,90 @@ void test7218()
 }
 
 /*******************************************/
+// 7517
+
+void test7517()
+{
+    static string result;
+
+    interface I
+    {
+        static I self;
+
+        void setEnable()
+        in
+        {
+            assert(self is this);
+            result ~= "I.setEnable.in/";
+            assert(!enabled);
+        }
+        out
+        {
+            assert(self is this);
+            result ~= "I.setEnable.out/";
+            assert( enabled);
+        }
+
+        void setDisable()
+        in
+        {
+            assert(self is this);
+            result ~= "I.setDisable.in/";
+            assert( enabled);
+        }
+        out
+        {
+            assert(self is this);
+            result ~= "I.setDisable.out/";
+            assert(!enabled);
+        }
+
+        @property bool enabled() const;
+    }
+
+    class C : I
+    {
+        static C self;
+
+        void setEnable()
+        in {}       // supply in-contract to invoke I.setEnable.in
```

Comment:
[This](https://github.com/D-Programming-Language/dmd/pull/4200) PR tries to make it so no in-contract would mean inherit from base and empty in-contract - relax the contract to not require anything, which I think is more logical.